pkg/controller/scylladbdatacenter/resource.go

```diff
@@ -1198,7 +1226,80 @@ func sysctlInitContainer(sdc *scyllav1alpha1.ScyllaDBDatacenter, image string) (
 			"-c",
 			fmt.Sprintf("sysctl -w %s", strings.Join(sysctls, " ")),
 		},
-	}, nil
+	}, true, nil
+}
+
+func makeScyllaDBBootstrapBarrierInitContainer(sdc *scyllav1alpha1.ScyllaDBDatacenter, image string) (*corev1.Container, bool, error) {
+	if !utilfeature.DefaultMutableFeatureGate.Enabled(features.BootstrapSynchronisation) {
+		return nil, false, nil
+	}
+
+	scyllaDBVersion, err := naming.ImageToVersion(sdc.Spec.ScyllaDB.Image)
+	if err != nil {
+		return nil, false, fmt.Errorf("can't get version of image %q: %w", sdc.Spec.ScyllaDB.Image, err)
+	}
+	sv := semver.NewScyllaVersion(scyllaDBVersion)
+	if !sv.SupportFeatureUnsafe(semver.ScyllaDBVersionRequiredForBootstrapSynchronisation) {
+		klog.V(4).InfoS("Not including bootstrap barrier init container as ScyllaDB version does not support it", "ScyllaDBDatacenter", naming.ObjRef(sdc), "ScyllaDBVersion", scyllaDBVersion, "ScyllaDBVersionRequiredForBootstrapSynchronisation", semver.ScyllaDBVersionRequiredForBootstrapSynchronisation)
+		return nil, false, nil
+	}
+
+	c := &corev1.Container{
+		Name:            "scylladb-bootstrap-barrier",
+		Image:           sdc.Spec.ScyllaDB.Image,
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		Command: []string{
+			"/mnt/shared/scylla-operator",
```

Comment:
Again, we could comment here why we expect the scylla-operator binary to be present in this directory while we run the ScyllaDB image.